src/platform_linux/platform.c

```diff
@@ -202,6 +202,140 @@ platform_spinlock_destroy(platform_spinlock *lock)
    return CONST_STATUS(ret);
 }
 
+/*
+ *-----------------------------------------------------------------------------
+ * Batch Read/Write Locks
+ *
+ * These are generic distributed reader/writer locks with an intermediate claim
+ * state. These offer similar semantics to the locks in the clockcache, but in
+ * these locks read locks cannot be held with write locks.
+ *
+ * These locks are allocated in batches of PLATFORM_CACHELINE_SIZE / 2, so that
+ * the write lock and claim bits, as well as the distributed read counters, can
+ * be colocated across the batch.
+ *-----------------------------------------------------------------------------
+ */
+
+void
+platform_batch_rwlock_init(platform_batch_rwlock *lock)
+{
+   ZERO_CONTENTS(lock);
+}
+
+/*
+ *-----------------------------------------------------------------------------
+ * lock/unlock
+ *
+ * Drains and blocks all gets (read locks)
+ *
+ * Caller must hold a claim
+ * Caller cannot hold a get (read lock)
+ *-----------------------------------------------------------------------------
+ */
+
+void
+platform_batch_rwlock_lock(platform_batch_rwlock *lock, uint64 lock_idx)
+{
+   platform_assert(lock->write_lock[lock_idx].claim);
```

Comment:
It will be useful to peel this one-line off into its own 
`platform_batch_rwlock_claimheld(platform_batch_rwlock *lock, uint64 lock_idx)`
inline function which implements the rule doc'ed on L231.

As this interface grows, we will find other places where "claim-held" rule will need to be checked.